src/environments/array.js

```diff
@@ -122,6 +130,9 @@ function parseArray(
         row.push(cell);
         const next = parser.fetch().text;
         if (next === "&") {
+            if (maxNumCols && row.length === maxNumCols) {
+                throw new ParseError("Too many tab characters: &");
```

Comment:
```suggestion
                throw new ParseError("Too many tab characters: &",
                                     parser.nextToken);
```
`ParseError` can provide position information if token is given as the second argument.